### DISCUSSION TITLE
Testing with Visual Studio Code

### DISCUSSION THREAD

**Discussion by mstniy** [isar/isar#352] [Help]
My team is trying to run tests for our application in Visual Studio Code, but keeps running into an error where Isar cannot be initialized as the app cannot find the relevant shared library:

```
IsarError: Could not initialize IsarCore library. If you create a Flutter app, make sure to add isar_flutter_libs to your dependencies.
```

Note that we do have the mentioned package as a dependency, and the database does indeed initialize correctly for the Android build, but we are having difficulty running tests. Using `flutter test` on the command line works, if the path to the shared library is provided in the `LD_LIBRARY_PATH` environment variable, but we have not been able to achieve this using vscode, and this makes it very difficult to debug our tests.

[The sample app](https://github.com/isar/samples) doesn't have any tests, either. What is the recommended way to have tests with Isar and vscode?

**Comment by scottbisaillon** [comment]
I have been successful in setting up tests using the method I have mentioned in this issue:

https://github.com/isar/isar/issues/294#issuecomment-1079234654

By going this route, I have been able to run tests through the CLI and through VS code.
